server/manager/src/main/java/org/apache/accumulo/manager/upgrade/UpgradeCoordinator.java

```diff
@@ -309,19 +305,19 @@ public UpgradeStatus getStatus() {
       justification = "Want to immediately stop all manager threads on upgrade error")
   private void abortIfFateTransactions(ServerContext context) {
     try {
-      final ReadOnlyFateStore<UpgradeCoordinator> mfs = new MetaFateStore<>(
-          context.getZooKeeperRoot() + Constants.ZFATE, context.getZooReaderWriter());
-      final ReadOnlyFateStore<UpgradeCoordinator> ufs = new UserFateStore<>(context);
-      try (var mfsList = mfs.list(); var ufsList = ufs.list();
-          var idStream = Stream.concat(mfsList, ufsList)) {
-        if (idStream.findFirst().isPresent()) {
-          throw new AccumuloException("Aborting upgrade because there are"
-              + " outstanding FATE transactions from a previous Accumulo version."
-              + " You can start the tservers and then use the shell to delete completed "
-              + " transactions. If there are incomplete transactions, you will need to roll"
-              + " back and fix those issues. Please see the following page for more information: "
-              + " https://accumulo.apache.org/docs/2.x/troubleshooting/advanced#upgrade-issues");
-        }
+      // The current version of the code creates the new accumulo.fate table on upgrade, so no
+      // attempt is made to read it here. Attempting to read it this point would likely cause a hang
+      // as not tablets are assigned when this is called. The Fate code is not used to read from
+      // zookeeper below because the serialization format changed in zookeeper, that is why a direct
+      // read is performed.
+      if (!context.getZooReader().getChildren(context.getZooKeeperRoot() + Constants.ZFATE)
```

Comment:
Ah was briefly looking over this and was wondering why it was only looking at ZK... This makes sense good catch. Haven't looked at the other classes but the changes for this class LGTM

Reply:
The upgrade from 4.0 to 4.1 will probably need to look at the new fate table, which will be tricky.  Will  open an issue for that now so its not forgotten.

Reply:
Opened #4721